### ISSUE TITLE
steamplotting for arbitrary input grids

### ISSUE THREAD

**Issue by DamienIrving** [SciTools/iris#1720]
For input data where the longitude values span -180 to 180, the following works fine for any output projection:
```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

input_projection = ccrs.PlateCarree()
ax = plt.subplot(nrows, ncols, plotnum, projection=output_projection)
ax.streamplot(x, y, u, v, transform=input_projection)
```

I've found that problems start to arise, however, as soon as the input projection isn't `ccrs.PlateCarree()` (i.e. when the longitude values aren't -180 to 180). For instance, if the input data has longitude values from 0 to 360 and I set `input_projection = ccrs.PlateCarree(central_longitude=180.0)`, the resulting streamline plot is shifted 180 degrees from the correct location (and if I try to set `output_projection = ccrs.PlateCarree(central_longitude=180.0)` only the western hemisphere shows data).

I got around this problem by using `new_cube = cube.intersection(longitude=(-180, 180, False, True))` to convert my input data to -180 to 180, however I thought I'd flag this issue as I'm assuming it was the intention to be able to draw streamlines for any arbitrary input projection.

**Comment by ajdawson** [comment]
This should really be a cartopy issue because iris doesn't properly support vector plotting at all yet. I think you'll get a more effective response if you move this issue over to the cartopy project.

There was a similar (cartopy) issue raised for quiver and barbs which ended up being a bug in matplotlib, and was fixed in matplotlib 1.4.x SciTools/cartopy#351. I don't think this is the same thing though, but I also don't know if it is a cartopy problem (streamplot requires interpolation before plotting) or another matplotlib problem.

**Comment by DamienIrving** [comment]
Thanks, @ajdawson. I've created a [new issue](https://github.com/SciTools/cartopy/issues/644) over at the Cartopy repo.
